jmsSource/src/main/java/io/vantiq/extsrc/jmsSource/communication/JMSMessageProducer.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019 Vantiq, Inc.
+ *
+ * All rights reserved.
+ *
+ * SPDX: MIT
+ */
+
+package io.vantiq.extsrc.jmsSource.communication;
+
+import javax.jms.BytesMessage;
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.DeliveryMode;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.MapMessage;
+import javax.jms.Message;
+import javax.jms.MessageProducer;
+import javax.jms.ObjectMessage;
+import javax.jms.Session;
+import javax.jms.StreamMessage;
+import javax.jms.TextMessage;
+import javax.naming.Context;
+import javax.naming.NamingException;
+
+public class JMSMessageProducer {
+    
+    public String destName;
+    
+    private Context context;
+    private ConnectionFactory connectionFactory;
+    private Connection connection;
+    private Session session;
+    private Destination destination;
+    private MessageProducer producer;
+    
+    public static final String MESSAGE = "message";
+    public static final String BYTES = "bytes";
+    public static final String TEXT = "text";
+    public static final String STREAM = "stream";
+    public static final String MAP = "map";
+    public static final String OBJECT = "object";
+    
+    public JMSMessageProducer(Context context) {
+        this.context = context;
+    }
+    
+    /**
+     * A method used to setup the MessageProducer for the given destination (topic or queue)
+     * @param connectionFactoryName     The name of the connection factory used to connect to the JMS Server
+     * @param dest                      The name of the destination to connect to (topic or queue)
+     * @param isQueue                   A boolean flag used to create the appropriate type of destination (queue or topic)
+     * @param username                  The username used to create the JMS Connection, (or null if JMS Server does not require auth)
+     * @param password                  The password used to create the JMS Connection, (or null if JMS Server does not require auth)
+     * @throws NamingException
+     * @throws JMSException
+     */
+    public void setupMessageProducer(String connectionFactoryName, String dest, boolean isQueue, String username, String password) throws NamingException, JMSException {
+        this.destName = dest;
+        connectionFactory = (ConnectionFactory) context.lookup(connectionFactoryName);
```

Comment:
same set of comments about error handling as in `JMSMessageListener`

Thus super/sub-class comment.  Naming this `open()` makes that easier...